### DISCUSSION TITLE
Warnings about which-key after latest updates

### DISCUSSION THREAD

**Discussion by SargisPlusPlus** [LazyVim/LazyVim#4008] [Q&A] [answered]
I am getting warnings:

```
There were issues reported with your **which-key** mappings.
Use `:checkhealth which-key` to find out more.
```

The actual issue is:
```
- WARNING You're using an old version of the which-key spec.
  Your mappings will work, but it's recommended to update them to the new version.
  Please check the docs and suggested spec below for more info.
  Mappings: >
```

While this isn't causing issues, I don't like see these everytime I start vim

**Comment by folke** [comment, marked as answer]
Should be fixed in the latest LazyVim version

**Reply by gtheys** [reply]
I just tried the latest starter and it still has the error. Am I doing something wrong?

**Reply by marioy47** [reply]
Same here

**Reply by ELIAWrecker** [reply]
same here

**Reply by nikita-edel** [reply]
it still happens with the newest version, at least for me
you can disable the warnings for which-key specifically, since which-key will work anyways (see my reply)

**Reply by dpetka2001** [reply]
You probably use old syntax which-key in your personal configuration and you're seeing this. I don't have any problems. And just now tested a fresh LazyVim installation without user configuration and it doesn't have any problem. 

Check your personal configuration and which-key docs.

**Comment by dpetka2001** [comment]
Just moments ago a new LazyVim stable version was released, so you shouldn't be getting any more warnings except if you have some kind of `which-key` in your personal configuration, in which case you should just change it according to the new syntax.

**Comment by thomassantosh** [comment]
There are some syntax changes that might need updating... I was using the kickstart init.lua file and updated the which-key plugin configuration to the [latest one](https://github.com/nvim-lua/kickstart.nvim/blob/186018483039b20dc39d7991e4fb28090dd4750e/init.lua#L274). This seems to have resolved this error.

**Reply by 0xlukaC** [reply]
changing .register to .add as the .register is now depreciated. 
```lua
 require('which-key').register {...
```
to 
```lua
 require('which-key').add {...
```

**Reply by chandragie** [reply]
this fixed mine! thanks

**Comment by nikita-edel** [comment]
paste this at the top of init.lua, (in case the warnings are meaning less/which-key still works)
i made a lot of research and havent found a better solution, it still happens to date on all versions for me

```lua
-- filter which-key warnings
local orig_notify = vim.notify
vim.notify = function(msg, level, opts)
  if msg:match("which%-key") and level == vim.log.levels.WARN then
    return
  end
  orig_notify(msg, level, opts)
end
